xbmc/games/controllers/input/PortManager.h

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright (C) 2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "games/controllers/types/ControllerTree.h"
+
+class TiXmlElement;
+
+namespace KODI
+{
+namespace GAME
+{
+class CPortManager
+{
+public:
+  CPortManager();
+  ~CPortManager();
+
+  void Initialize(const std::string& profilePath);
+  void Deinitialize() {}
```

Comment:
Is this just copy/paste?

Reply:
A combination of laziness and forgetfulness? I pushed an implementation.